### PR TITLE
Update supported platforms topic for 0.41.0

### LINKED ISSUE #1175: Update supported platforms topic for 0.41.0
Update https://www.eclipse.org/openj9/docs/openj9_support/ with changes for 0.41.0, including changes for Java 21.

- Update the "Eclipse OpenJ9 releases" table. Remove the note (1) saying that 0.41 is a future release. Add more future releases and remove older ones.
- Any other platform changes
- any changes to compiler levels
- A separate table required for OpenJDK 21?
- Any other changes in this document

### PR DESCRIPTION
**Pull request by Sreekala-Gopakumar**
https://github.com/eclipse-openj9/openj9-docs/issues/1175

Updated the Eclipse OpenJ9 releases table.

Closes #1175
Signed-off-by: Sreekala Gopakumar <sreekala.gopakumar@ibm.com>

### CODE REVIEW

**Comment by Sreekala-Gopakumar** [comment]
Jenkins doc stage

File: docs/openj9_support.md [hunk 1]
```diff
@@ -42,12 +42,14 @@ columns will be removed over time.
...
+| 0.41.0         | Oct 2023   |  :fontawesome-solid-check:{: .yes aria-hidden="true"}<span class="sr-only">yes</span> | :fontawesome-solid-check:{: .yes aria-hidden="true"}<span class="sr-only">yes</span> | :fontawesome-solid-check:{: .yes aria-hidden="true"}<span class="sr-only">yes</span>     |  :fontawesome-solid-xmark:{: .no aria-hidden="true"}<span class="sr-only">no</span>          |        |
+| 0.42.0         | Oct 2023 (1, 3)   |  :fontawesome-solid-xmark:{: .no aria-hidden="true"}<span class="sr-only">no</span> | :fontawesome-solid-xmark:{: .no aria-hidden="true"}<span class="sr-only">no</span> | :fontawesome-solid-xmark:{: .no aria-hidden="true"}<span class="sr-only">no</span>     |  :fontawesome-solid-xmark:{: .no aria-hidden="true"}<span class="sr-only">no</span>          |  :fontawesome-solid-check:{: .yes aria-hidden="true"}<span class="sr-only">yes</span>       |
```
Comment (pshipton) [inline docs/openj9_support.md:51]: Oct seems unlikely, lets use Dec.

**Comment by Sreekala-Gopakumar** [comment]
Jenkins doc stage

**Comment by Sreekala-Gopakumar** [comment]
@pshipton - I have updated the table. Please merge if ok. Thanks!

**Comment by pshipton** [comment]
Pls squash the commits so we don't have commits to add and revert zlibnx changes.

**Comment by pshipton** [comment]
There are still 2 commits (one for zlibnx) but there should, just be the one.

**Comment by pshipton** [comment]
Create a new branch from master, cherry pick the one commit here that we want, and then force push the new branch to the `1175UpdtSupEnv ` branch in your remote fork.

**Comment by Sreekala-Gopakumar** [comment]
Jenkins doc stage

**Comment by Sreekala-Gopakumar** [comment]
@pshipton - Done.

**Review by pshipton — APPROVED** [review]

**Pull request merged**